source/synthDrivers/sapi4.py

```diff
@@ -15,20 +15,26 @@
 from ._sapi4 import *
 import config
 import nvwave
+import weakref
+
 
 class SynthDriverBufSink(COMObject):
 	_com_interfaces_ = [ITTSBufNotifySink]
 
-	def __init__(self,synthDriver):
-		self._synthDriver=synthDriver
+	def __init__(self, synthDriver):
+		self.synthRef = weakref.ref(synthDriver)

```

Comment:
Rather than making this a weak ref here, I think it would be a better design to pass in a weakref. 

We should make the design of this clearly state the intentions for lifetime management of objects. So either, this takes a synthDriver object, and is responsible for it's lifetime (the parent relinquishes control), or the parent does not give the indication of control, instead this takes a `weakref[synthdriver]`. If memory serves, in the case of #9964, I think in a roundabout way the synthDriver owned the sink, causing a circular reference. 

What are your thoughts on this?

Note, I think that type annotations struggle with weakref unfortunately.